plugins/formal-verification/testData/diagnostics/good_contracts/returns_booleans.fir.diag.txt

```diff
@@ -77,12 +77,14 @@ method global$fun_binary_logic_expressions$fun_take$T_Boolean$T_Boolean$return$T
 
 
 /returns_booleans.kt:(1268,1281): info: Generated Viper text for isNullOrEmpty:
-method pkg$kotlin$collections$class_scope_pkg$kotlin$collections$global$class_Collection$getter_size(this: Ref)
-  returns (ret: Int)
-
-
 method global$fun_isNullOrEmpty$fun_take$NT_class_pkg$kotlin$collections$global$class_Collection$return$T_Boolean(this: dom$Nullable[Ref])
   returns (ret: Bool)
+  requires acc((dom$Casting$cast(this, dom$Type$pkg$kotlin$collections$global$class_Collection()): Ref).special$size, write)
```

Comment:
I just realised, this is not actually correct.  This access predicate should hold under the condition that `this` is not null, so I think you need to use an implication here.